src/test/java/com/example/employee/controller/EmployeeControllerTest.java

```diff
@@ -54,8 +75,111 @@ void getAllEmployees() throws Exception {
         EmployeeDTO expectedEmployeeDto = expectedEmployeeDTOList.stream().findFirst().orElseThrow();
         EmployeeDTO actualEmployeeDto = actualEmployeeDTOList.stream().findFirst().orElseThrow();
 
-        Assertions.assertEquals(expectedEmployeeDto.getFirst_name(), actualEmployeeDto.getFirst_name());
-        Assertions.assertEquals(expectedEmployeeDto.getLast_name(), actualEmployeeDto.getLast_name());
+        assertThat(expectedEmployeeDto)
+                .usingRecursiveComparison()
+                .ignoringFields("id")
+                .isEqualTo(actualEmployeeDto);
     }
 
+
+    @Test
+    void getEmployeeById() throws Exception {
+        EmployeeDTO employeeDTO = new EmployeeDTO.EmployeeDTOBuilder()
+                .first_name("Nika")
+                .last_name("Avalishvili")
+                .department("Business development")
+                .positions("Business builder")
+                .email("avalishvili.nick@gmail.com")
+                .is_active(true)
+                .is_pensions_payer(true)
+                .buildEmployeeDTO();
+        Long id = employeeService.createAndUpdateEmployee(employeeDTO).getId();
+
+        String responseAsAString = mockMvc.perform(MockMvcRequestBuilders.get("/employee/{id}", id))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString();
+
+        EmployeeDTO actualEmployeeDTO = objectMapper.readValue(responseAsAString, new TypeReference<>() {
+        });
+
+        assertThat(actualEmployeeDTO)
+                .usingRecursiveComparison()
+                .ignoringFields("id")
+                .isEqualTo(employeeDTO);
+    }
+
+    @Test
+    void addOrUpdateEmployee() throws Exception {
+        EmployeeDTO employeeDTO = new EmployeeDTO.EmployeeDTOBuilder()
+                .first_name("Harry")
+                .last_name("Potter")
+                .department("Ministry of Magic")
+                .positions("Clerk")
+                .email("Harry.potter@ministryofmagic.com")
+                .is_active(true)
+                .is_pensions_payer(false)
+                .buildEmployeeDTO();
+        employeeService.createAndUpdateEmployee(employeeDTO);
+
+        objectMapper.configure(SerializationFeature.WRAP_ROOT_VALUE, false);
+        ObjectWriter ow = objectMapper.writer().withDefaultPrettyPrinter();
+        String requestJson = ow.writeValueAsString(employeeDTO);
```

Comment:
Please, check if we need this configuration
Maybe, we can just use `objectMapper.writeValueAsString(employeeDTO)` here